azurecompute-arm/src/test/java/org/jclouds/azurecompute/arm/features/VaultApiLiveTest.java

```diff
@@ -0,0 +1,1220 @@
+           "L/xiHRpyzec/jrxJxAIIwFHiw2jbLdxNqDaVPFA6X1Cks4fvY40KymOS8Ecmkgx4\n" +
+           "C6/ZqLECgYEA38rL8zmbJRRWgrxSoROucaN/9DyvE8Hd97s1zf3I0LIF+GI3JdcN\n" +
+           "DV5O5VDIgQ7QZ55lOaTqJ0f2fOQZF1UbTU1gBUHVF6j1nv6Xic3OV+ZmhTMbt0Op\n" +
+           "WxPaKup6dkNEAe/Xg0p308r8xw/psh4/gjL1ICHwycjUlz9FQz8FLGsCgYEA2OHc\n" +
+           "/F4vAdK04U4w6AyivuJdIsDAks1ikl+3TqgzyesPg+DpKVNSy6DhraDFKdRqWhbK\n" +
+           "DqigTud8TVk9kmyF3WIb1BDf4IrxlTK08s6Jf25QA/lBlwIst3rejqWwRBY2fJp4\n" +
+           "O8hU31xNLd8nZq8tKYaP+yvyI6fSC+9ZIgyATl8CgYBtTlYzZG2cvMRMcsNBHaXU\n" +
+           "p3E1z/YLhmNuPqhXBp/xHyK/YOliuBkN8IREiSKFtsCf+8OhViVwLjv8YETJGq6N\n" +
+           "IT/HEZKNTd86W0uU6UrhsA1e/cmJx6iObbHfyEssPEqZT5ZJ8DN65Oh2KhWRBCks\n" +
+           "5MrJeQps5ITBowunfr5+JQKBgBln09goPD9tbEzcOPRlzoanw5s3kxdfQcW9up5X\n" +
+           "K8HibC3ZodZ6cT9zOBx7sbzlqhOlicPP0v+LNE9nPcQyPDgjrvjXhTIrdIMyrmmm\n" +
+           "8gpgZLUidKF42r4fgYSaOhyvsXBFNAJkxDRp/ox6EIaBzLuJjMIycqnklcFaJ0nK\n" +
+           "xxjnAoGAU/3jRok74Yn0MY183UCXICf/WxVYzwEi3cyRWVh6gAo2ExNye97rau/B\n" +
+           "Y2woAcVxEN9h1HNirq8Z2AwrZKd/BcBnvulZqECeH5zBQ0W56ADX72r+lzOAxTJa\n" +
+           "U1AIlNCIkfsQ5CpIoYsVoNEM8P+GyS8B9kaEmk/FVKldRTKJqkI=\n" +
+           "-----END RSA PRIVATE KEY-----";
+   private static String cryptoText = "R29sZCUyNTIxJTJCR29sZCUyNTIxJTJCR2" +
+           "9sZCUyQmZyb20lMkJ0aGUlMkJBbWVyaWNhbiUyQlJpdmVyJTI1MjE";
+   private static String cryptoAlgorithm = "RSA-OAEP";
+   private static String hashToSign = "FvabKT6qGwpml59iHUJ72DZ4XyJcJ8bgpgFA4_8JFmM";
+   private static String signatureAlgorithm = "RS256";
+   private static String contentEncryptionKey = "YxzoHR65aFwD2_IOiZ5rD08jMSALA1y7b_yYW0G3hyI";
+
+   @BeforeClass
+   @Override
+   public void setup() {
+      super.setup();
+      createTestResourceGroup();
+      vaultName = String.format("kv%s", this.getClass().getSimpleName().toLowerCase());
+   }
+
+   @AfterClass
+   public void forceVaultRemoval() {
+      // see if the vault has been deleted or not
+      Vault vault = api().getVault(vaultName);
+      if (vault != null) {
+         if ((vault.properties().enableSoftDelete() != null) && vault.properties().enableSoftDelete()) {
+            api().deleteVault(vaultName);
+            waitForOperation("vault", "delete", vaultName);
+         } else {
+            return;
+         }
+      }
+
+      DeletedVault deletedVault = api().getDeletedVault(LOCATION, vaultName);
+      if (deletedVault != null) {
+         api().purgeVault(LOCATION, vaultName);
+         waitForOperation("vault", "purge", vaultName);
+      }
+   }
+
+   // Note:  Operations on a KeyVault that supports soft delete can take time
+   // to settle.  Thus for tests where appropriate, a wait operation with a
+   // one minute timeout will be used to make sure things are in the desired
+   // state before proceeding to the next test.
+   private boolean waitForOperation(String itemType, String operation, String entityName) {
```

Comment:
jclouds already provides clean means to actively wait for things, and allows to configure the timeouts, etc, via properties. Please use the `Predicates2.retry` helper instead of manually doing this.
This logic might be useful for users too, so I'd strongly recommend to copy how [we configure active waits for Azure ARM](https://github.com/jclouds/jclouds-labs/blob/master/azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/compute/config/AzureComputeServiceContextModule.java#L145-L385) and apply the same pattern here:

1. Refactor each conditional in this method into its own `Predicate` class.
2. Put all those predicates in an `AzurePredicatesModule` (or similar), and consider moving the ones that already exist (linked above) to that class.
3. For each defined predicate class, add a `@Provides` method that exposes the predicate encapsulated with the `retry` helper. Consider using the standard jclouds timeout properties or just create Azure-specific properties to configure them if that makes sense, but follow the same pattern.
4. [Initialize the predicates in the base test class](https://github.com/jclouds/jclouds-labs/blob/master/azurecompute-arm/src/test/java/org/jclouds/azurecompute/arm/internal/BaseAzureComputeApiLiveTest.java#L106-L112) so you can easily use them in the tests.